storage/src/backend/registry.rs

```diff
@@ -136,7 +136,7 @@ enum Auth {
 
 struct RegistryState {
     // HTTP scheme like: https, http
-    scheme: String,
+    scheme: RwLock<String>,
```

Comment:
It would be better to use `AtomicBool` instead of `RwLock<String>`, to avoid acquiring lock on each request.

Reply:
We need to change the value of the scheme. I am confused about how to use `AtomicBool`

Reply:
> We need to change the value of the scheme. I am confused about how to use `AtomicBool`

use `is_https: AtomicBool` as a flag to choose https or http. And we can access/change value of `is_https` without locking:)

Reply:
Maybe use `type Scheme = AtomicBool<true>` as `https`, `type Scheme = AtomicBool<false>` as `http`, then implement `Display` for `Scheme` type.

Reply:
@jiangliu @uran0sH Note: if users have specified the `scheme` field, for performance consideration, we shouldn't automatically fallback to `http`.